Author: Michael Stapelberg <stapelberg@debian.org>
Description: Open the PAM session in the parent process.
 This is necessary when logind is running. pam_systemd communicates with logind
 and sets up a file descriptor that represents the session. When this file
 descriptor is closed, logind will close the session and kill all processes as
 part of that. The file descriptor is marked CLOEXEC, so without this patch,
 the file descriptor gets opened, then wdm exec()s the session and the file
 descriptor is closed, leading to logind immediately killing the session.
 This patch makes wdm do PAM session handling the same way that xdm does it.
Last-Updated: 2013-08-28
Forwarded: not-needed, project is dead upstream

Index: wdm-1.28/src/wdm/session.c
===================================================================
--- wdm-1.28.orig/src/wdm/session.c	2005-03-26 14:57:04.000000000 +0100
+++ wdm-1.28/src/wdm/session.c	2013-08-28 22:08:43.819028300 +0200
@@ -550,6 +550,7 @@
 	WDMDebug("\n");
     }
 #ifdef USE_PAM
+    if (pamh) pam_open_session(pamh, 0);
 #endif    
     switch (pid = fork ()) {
     case 0:
@@ -599,8 +600,6 @@
 		for(i = 0; pam_env && pam_env[i]; i++) {
 			verify->userEnviron = WDMPutEnv(verify->userEnviron, pam_env[i]);
 		}}
-
-		pam_open_session(pamh, 0);
 	}
 #endif
 	if (setuid(verify->uid) < 0)
